src/main/java/com/nutomic/syncthingandroid/syncthing/RestApi.java

```diff
@@ -440,9 +440,14 @@ protected void onPostExecute(String s) {
                     SystemInfo si = new SystemInfo();
                     si.alloc = system.getLong("alloc");
                     si.cpuPercent = system.getDouble("cpuPercent");
-                    si.extAnnounceTotal = system.getInt("discoveryMethods");
-                    si.extAnnounceConnected =
+                    if system.has("discoveryEnabled") {
```

Comment:
Java requires parentheses around the if condition: `if (system.has("discoveryEnabled"))`. Add that and I'll merge it ;)